extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -844,6 +860,24 @@ export class AmpA4A extends AMP.BaseElement {
     });
   }
 
+  renderViaNameFrame_(creativeBody) {
+    this.emitLifecycleEvent('renderSafeFrameStart');
+    return utf8Decode(creativeBody).then(creative => {
+      /** @const {!Element} */
+      const name_data = Object.create(null);
```

Comment:
Why not just const name_data = {'creative': creative}?

Also a decent amount of duplication between this and safeframe code... you could have a single render via frame loader function that takes just the creative and then uses this.experimentalNonAmpCreativeRenderMethod_ to determine which (or you can pass the value if you're worried about state).